charts/logzio-metrics-collector/templates/_helpers.tpl

```diff
@@ -0,0 +1,225 @@
+{{- end -}}
+
+
+{{/*
+Compute InternalTrafficPolicy on Service creation
+*/}}
+{{- define "metrics-collector.serviceInternalTrafficPolicy" }}
+  {{- if and (eq .Values.mode "daemonset") (eq .Values.service.enabled true) }}
+    {{- print (.Values.service.internalTrafficPolicy | default "Local") -}}
+  {{- else }}
+    {{- print (.Values.service.internalTrafficPolicy | default "Cluster") -}}
+  {{- end }}
+{{- end -}}
+
+{{/*
+Allow the release namespace to be overridden
+*/}}
+{{- define "metrics-collector.namespace" -}}
+  {{- if .Values.namespaceOverride -}}
+    {{- .Values.namespaceOverride -}}
+  {{- else -}}
+    {{- .Release.Namespace -}}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+  This helper converts the input value of memory to Bytes.
+  Input needs to be a valid value as supported by k8s memory resource field.
+ */}}
+{{- define "metrics-collector.convertMemToBytes" }}
+  {{- $mem := lower . -}}
+  {{- if hasSuffix "e" $mem -}}
+    {{- $mem = mulf (trimSuffix "e" $mem | float64) 1e18 -}}
+  {{- else if hasSuffix "ei" $mem -}}
+    {{- $mem = mulf (trimSuffix "e" $mem | float64) 0x1p60 -}}
+  {{- else if hasSuffix "p" $mem -}}
+    {{- $mem = mulf (trimSuffix "p" $mem | float64) 1e15 -}}
+  {{- else if hasSuffix "pi" $mem -}}
+    {{- $mem = mulf (trimSuffix "pi" $mem | float64) 0x1p50 -}}
+  {{- else if hasSuffix "t" $mem -}}
+    {{- $mem = mulf (trimSuffix "t" $mem | float64) 1e12 -}}
+  {{- else if hasSuffix "ti" $mem -}}
+    {{- $mem = mulf (trimSuffix "ti" $mem | float64) 0x1p40 -}}
+  {{- else if hasSuffix "g" $mem -}}
+    {{- $mem = mulf (trimSuffix "g" $mem | float64) 1e9 -}}
+  {{- else if hasSuffix "gi" $mem -}}
+    {{- $mem = mulf (trimSuffix "gi" $mem | float64) 0x1p30 -}}
+  {{- else if hasSuffix "m" $mem -}}
+    {{- $mem = mulf (trimSuffix "m" $mem | float64) 1e6 -}}
+  {{- else if hasSuffix "mi" $mem -}}
+    {{- $mem = mulf (trimSuffix "mi" $mem | float64) 0x1p20 -}}
+  {{- else if hasSuffix "k" $mem -}}
+    {{- $mem = mulf (trimSuffix "k" $mem | float64) 1e3 -}}
+  {{- else if hasSuffix "ki" $mem -}}
+    {{- $mem = mulf (trimSuffix "ki" $mem | float64) 0x1p10 -}}
+  {{- end }}
+{{- $mem }}
+{{- end }}
+
+{{- define "metrics-collector.gomemlimit" }}
+{{- $memlimitBytes := include "metrics-collector.convertMemToBytes" . | mulf 0.8 -}}
+{{- printf "%dMiB" (divf $memlimitBytes 0x1p20 | floor | int64) -}}
+{{- end }}
+
+
+{{/*
+Calculate Logz.io listener address based on region
+*/}}
+{{- define "logzio.listenerAddress" -}}
+{{- $region := .Values.secrets.logzioRegion -}}
+{{- if eq $region "us" -}}
+https://listener.logz.io:8053
+{{- else if eq $region "au" -}}
+https://listener-au.logz.io:8053
+{{- else if eq $region "ca" -}}
+https://listener-ca.logz.io:8053
+{{- else if eq $region "eu" -}}
+https://listener-eu.logz.io:8053
+{{- else if eq $region "uk" -}}
+https://listener-uk.logz.io:8053
+{{- else -}}
+https://listener.logz.io:8053  # Default to us-east-1 region if no match
+{{- end -}}
+{{- end }}
```

Comment:
I think this can be shortened:
```
{{- define "logzio.listenerAddress" -}}
{{- $region := .Values.secrets.logzioRegion }}
{{- if or (eq $region "us") (not $region) }}
https://listener.logz.io:8053
{{- else }}
{{ printf "https://listener-%s.logz.io:8053" $region }}
{{- end }}
{{- end }}
```